exec/java-exec/src/main/java/org/apache/drill/exec/coord/zk/ZKClusterCoordinator.java

```diff
@@ -206,6 +205,16 @@ public DistributedSemaphore getSemaphore(String name, int maximumLeases) {
     return new ZkDistributedSemaphore(curator, "/semaphore/" + name, maximumLeases);
   }
 
+  @Override
+  public <V> TransientStore<V> newTransientStore(final TransientStoreConfig<V> config) {
+    final ZkEphemeralStore<V> store = new ZkEphemeralStore<>(config, curator);
```

Comment:
I think you need to cache these since each one will create a listener. This used to be done by cachingstoreprovider but I don't think that is in the Transient path anymore.